### ISSUE TITLE
FR: How to build the tailscale web interface, I hope it will be simpler

### ISSUE THREAD

**Issue by lmq8267** [tailscale/tailscale#9312]
### What are you trying to do?

I'm trying to build tailscale 1.49.0. This version requires the web interface to be built separately. 1.48.* No need to build web,
When building a web page, it prompts qt.qpa.xcb: could not connect to display   。 
There are still many errors in the construction process. Can you restore the web page to the previous way or make it easier to build the web?

Does anyone have a simpler construction method that can tell me?

**Comment by DentonGentry** [comment]
You are welcome to check out `release-branch/1.48` to build from the 1.48.x client release.

We will look into adding further build instructions on `main`

**Comment by lmq8267** [comment]
Thank you for your reply. Hope that future versions can be built as conveniently as 1.48.*.

**Comment by lmq8267** [comment]
Can it be like 1.48.* without building the web separately? If not, can you package the web and then use the command to decide whether to add the web when building taislcale?

**Comment by DentonGentry** [comment]
@lmq8267 We're updating `tailscale web` to use web technologies like React. This means it needs a build step to bundle JS dependencies. If you can include the error output it will help identify how to fix it.

**Comment by bradfitz** [comment]
The plan is to publish the compiled web output and have the tailscale Go binaries download it from the internet at runtime once as needed. But the official release binaries will embed it. Those parts aren't done yet.

**Comment by lmq8267** [comment]
Yes, you need to package the web. The process of packaging the web ourselves is too cumbersome and lacks many software and libs. If you package the web, the process of building tailscale will be much simpler. For example, 1.48.* can be easily built.

**Comment by bradfitz** [comment]
Yes, we know. Note that our version numbering is to use even minor numbers for stable releases (48 is even) and odd for dev cycle unstable builds (49). This is all a work in progress for a dev cycle.

**Comment by lmq8267** [comment]
Okay, it seems that I was too impatient and used the development version. I look forward to you building a good web, so that it will be much easier for us to build tailscale. Tailscale updates should be simpler and not more complex.

**Comment by soniaappasamy** [comment]
As of https://github.com/tailscale/tailscale/commit/652f77d23623c0e198338397c09afca41ce8e4d8, the web client assets are pulled into the tailscale repo from a prebuilt package. No additional steps are necessary anymore to run the web client off of main. Closing this ticket out.

**Comment by wfio** [comment]
This still shows up in both the unRaid CA plugin and Docker container

**Comment by bradfitz** [comment]
@wfio, it's fixed as of the latest 1.51.x dev builds, not in older builds.
